File: web/src/app/core/models/room.model.ts
import {Identity} from './identity';
import {Department} from './department.model';

export interface RoomModel {
  number: string;
  name: string;
  description: string;
  imageUrl?: string;
}

export interface Room extends RoomModel, Identity {}

export interface RichRoom extends RoomModel, Identity {
  department: Department;
}

export interface RoomForm extends RoomModel, Identity {
  department: string;
}

File: backend/src/main/kotlin/de/thkoeln/colab/roomsserver/core/models/RoomModel.kt
/*
 * Copyright (c) 2020  colab  TH-Köln
 * @author Florian Leonhard
 * All rights reserved
 */

package de.thkoeln.colab.roomsserver.core.models

interface RoomModel {
    val number: String
    val name: String
    val description: String
    val imageUrl: String?
}

interface RoomIdentity: RoomModel, Identity